Added OfficeMeta Service

ms-xy opened the pull request:
Requires 7832ab6b0d4d097c4aa3c309d6fc6335236cd0bb from the holmes-totem-library to be merged first. See HolmesProcessing/Holmes-Totem-Service-Library#2

ms-xy committed:
Added OfficeMeta Service

Requires 7832ab6b0d4d097c4aa3c309d6fc6335236cd0bb from the library to be merged.

Signed-off-by: Maximilian Schott <ms@roottec.com>

webstergd commented on config/totem.conf.example:
```diff
@@ -50,6 +50,10 @@ totem {
     }
+    office_meta {
+      uri = ["http://127.0.0.1:7730/office_meta/"]
+      resultRoutingKey = "office_meta.result.static.totem"
```
Please change office_meta and office_meta.result.static.totem to officemeta so the styles match.

webstergd commented on src/main/scala/org/novetta/zoo/driver/driver.scala:
```diff
@@ -4,6 +4,7 @@ import java.util.concurrent.{Executors, ExecutorService}
 
 import akka.actor.{ActorRef, ActorSystem, Props}
 import org.novetta.zoo.actors._
+import org.novetta.zoo.services.office_meta.{OfficeMetaSuccess, OfficeMetaWork}
```
same here. Just officemeta

webstergd commented on src/main/scala/org/novetta/zoo/driver/driver.scala:
```diff
@@ -126,9 +130,10 @@ object driver extends App with Instrumented {
 
     def workRoutingKey(work: WorkResult): String = {
       work match {
+        case x: MetadataSuccess => conf.getString("totem.enrichers.metadata.resultRoutingKey")
```
Thank you for making it alphabetical

webstergd commented on src/main/scala/org/novetta/zoo/services/office_meta/OfficeMetaREST.scala:
```diff
@@ -0,0 +1,42 @@
+package org.novetta.zoo.services.zipmeta
```
please fix the typo for the package and default routing keys.

webstergd commented on src/main/scala/org/novetta/zoo/services/office_meta/service.conf.example:
```diff
@@ -0,0 +1,11 @@
+Name        = OfficeMeta
+Version     = 1.0
+Description = ./README.md
+Copyright   = Copyright 2015 Holmes Processing
```
please change to 2016

webstergd commented on src/main/scala/org/novetta/zoo/services/office_meta/service.py:
```diff
@@ -0,0 +1,137 @@
+import tornado
```
Please add comments for each import block. Look at peinfo for an example.

webstergd commented:
Thanks dude. Please test. Additionally I would like the following changes: 
Please fix the typos, adjust to match the naming convention of the other services, provide a meaningful commit description, and add a comment on why you selected to use the CRITs' officemeta service. On your justification I want to know why you believe porting the CRITs' Service was a better verses the other available options or writing your own.

ms-xy committed:
Fixed some errors and obeying naming conventions now.

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy committed:
Cleaned up imports.

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy committed:
Adjusted licence copyright date.

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy committed:
Further cleanup on imports.

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy commented:
All changes as requested, sbt assembly runs through, sorry for the mistakes on the naming convention.
You might want to look at the cleanup I've done on the imports, maybe we should do that for other services using tornado as well.

ms-xy committed:
Fixed routing keys and work types.

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy commented:
As requested a short writeup as to why the CRITs implementation was chosen:
CRITs is state-of-the-art, widely accepted, their services are working and thus a reliable source.
Since we try to implement as much compatibility with CRITs as possible, it's good to start out with their implementation and incrementally improve it. (Implement support for newer office document types for example)

ms-xy committed:
Extensive changes and additions.

+ added ooxml/mox and xml parser (mso parser yet missing)
+ moved helper classes to respective files
+ cleaned up for readability

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy commented:
Extended it to now parse Office-XML (2003 version) as well as Office Open XML (Microsoft Office XML 2007) formatted files.
A new problem I see about this service now is that the return format can't be 100% conform with the previous format, as that one relied on the document structure of Microsofts OLE files.
I tried to keep it as close as possible for the new file types, but it isn't quite the same. :-1:

ms-xy committed:
Implemented ActiveMime/MSO decompression, corrected its usage.

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy committed:
Naming fix in docker-compose file for services

Signed-off-by: Maximilian Schott <ms@roottec.com>

webstergd commented:
what's the current status of this? are you still working it?

ms-xy commented:
For some reason I thought it was merged. I'll have to revisit it to see what the errors are, I haven't touched it in 4 months ...
I'll get to it later this week, thursday or friday.

ms-xy commented:
Pushing this one back a bit, as I'm currently still working on the PT service.

webstergd commented:
closing pull request until the code is ready to be submitted. When you are ready, please resubmit the pull request.
